### PR TITLE
profiling: Fix a data race in the circular buffer.

### PR DESCRIPTION
**Pull request by easwars**
Also fix linter errors.

### CODE REVIEW

File: internal/profiling/buffer/buffer.go [hunk 1]
```diff
@@ -255,8 +257,9 @@ func (cb *CircularBuffer) Drain() []interface{} {
 
 	result := make([]interface{}, 0)
 	for i := 0; i < len(qs); i++ {
-		if qs[i].acquired < qs[i].size {
-			result = dereferenceAppend(result, qs[i].arr, 0, qs[i].acquired)
+		acquired := atomic.LoadUint32(&qs[i].acquired)
```
Comment (dfawley) [inline internal/profiling/buffer/buffer.go]:
```
if acquired := atomic.LoadUint32(...); acquired < qs[i].size {
...
```

**Comment by dfawley** [comment]
(No release notes since we didn't do a release with profiling (which would also be internal anyway).)

**Comment by adtac** [comment]
LGTM :+1:

File: internal/profiling/buffer/buffer.go (hunk 1 above)
Comment (easwars) [reply internal/profiling/buffer/buffer.go]: Done

**Pull request merged**